generator/src/main/java/org/apache/kafka/message/FieldSpec.java

```diff
@@ -101,11 +101,7 @@ public FieldSpec(@JsonProperty("name") String name,
         this.entityType.verifyTypeMatches(name, this.type);
 
         this.about = about == null ? "" : about;
-        if (!this.fields().isEmpty()) {
-            if (!this.type.isArray()) {
-                throw new RuntimeException("Non-array field " + name + " cannot have fields");
-            }
-        }
+
```

Comment:
I don't think we should get rid of this check entirely.  We should change it to check for either array or struct type, rather than just array type, right?

It wouldn't make sense to have an int32 with fields, for example.

Reply:
Makes sense. Done.